src/classes/GS_ApplicationStatusController.cls

```diff
@@ -0,0 +1,176 @@
+                                                    : org.IsSandbox;
+    }
+    /**
+     * @description Retrieves the application date.
+     * On prod this is populated using the FeatureParameter "P10_Application_Date"
+     * For QA purpose this values can be overriden if this is running on a Scratch org and there is a value in the custom setting Testing_Configuration__c
+     * @param testingConfiguration Map<String, Object> Map of string, object with the testing configuration for overriden the returning values.
+     * @return   return Application date gotten of the P10_Application_Date FeatureParameter OR the 
+     */
+    private Date getApplicationDate(Map<String, Object> testingConfiguration) {
+        
+        if (isScratchOrg() && testingConfiguration != null) { 
+            return testingConfiguration.get(APPLICATION_PARAMETER) == null ? null: Date.parse(testingConfiguration.get(APPLICATION_PARAMETER).toString());
+        }
+
+        try {
+            return System.FeatureManagement.checkPackageDateValue(APPLICATION_DATE);
+        } catch (System.NoDataFoundException ex) {
+            return Date.Today();
+        }
+    }
+
+    /**
+     * @description Retrieves the Trial expiration date.
+     * On prod this is populated quering the Organization object.
+     * For QA purpose this values can be overriden if this is running on a Scratch org and there is a value in the custom setting Testing_Configuration__c
+     * @param  overrideValuesForTesting Boolean true if usage of Testing_Configuration__c custom setting is desired
+     * @param  testingConfiguration     Map<String, Object> parsed TestingConfiguration object
+     * @param  org                      Organization the Organization object, the query should have included the TrialExpiration field 
+     * @return                          DateTime with the expirationDate.
+     */
+    private DateTime getExpirationDate(Boolean overrideValuesForTesting,Map<String,Object> testingConfiguration, Organization org) {
+        
+        if (overrideValuesForTesting) {
+            return  (testingConfiguration.get(TRIAL_EXPIRATION_PARAMETER) == null ? null : 
+                    DateTime.parse(testingConfiguration.get(TRIAL_EXPIRATION_PARAMETER).toString()));
+        } else {
+            return org.TrialExpirationDate;
+        }
+        
+    }
+
+    /**
+     * @description Returns the parsed Testing_Configuration.
+     * @return   Map<String, Object> Parsed Testing_Configuration Custom_Setting. it is a json format.
+     * Examples: 
+     * {"Sandbox":false} simulates an Active org.
+     * {"Sandbox":false,"TrialExpiration":"10/15/2020, 00:01 am”} simulates a trial org which is not applied for licenses.
+     * {"Sandbox":false,"TrialExpiration":"10/15/2020","Application":"10/02/2020"} Simulates a trial org which has applied for licenses.
+     */
+    private Map<String, Object> getTestingConfiguration() {
```

Comment:
Is this object strictly used to simulate testing? If so, I'm not sure I'm comfortable with the addition of a custom object for that. Let's talk about other ways to accomplish this without building in a new (never-removable) custom object into the package.

Reply:
This is a Custom Setting. We add this after discussing with Marcel and David Reed. The component has different behavior according if the ORG is a Trial/Active and if the Application for the P10 licenses is available(this is coming from a new feature parameter). The idea is to give the QA a way to injects the values they need to test the paths we have.

Reply:
Thanks. I'll check in with David and Marcel on this. Let's hold the PR in the mean time.

Reply:
I'm thinking now, What do you think if instead of using the CustomSetting, we get the overriding data from a static resource which is not in the package? It will provide testeability now without putting in the package any new object that we wont be able to delete it in the future?